lib/puppet/provider/user/hpux.rb

```diff
@@ -19,14 +22,23 @@
     value !~ /\s/
   end
 
-  has_features :manages_homedir, :allows_duplicates, :manages_passwords
+  has_features :manages_homedir, :allows_duplicates, :manages_passwords, :manages_password_age, :manages_expiry
 
   def deletecmd
     super.insert(1,"-F")
   end
 
   def modifycmd(param,value)
-    super.insert(1,"-F")
+     cmd = super(param, value)
+     cmd << "-F"
+     if self.trusted == "Trusted"
+        cmd << ";"
```

Comment:
I think I just addressed that in
https://groups.google.com/d/msg/puppet-dev/MG-zwQYaKYM/IlXueb9pjbUJ.

On Wed, Jan 14, 2015 at 4:07 PM, jjperry69 <notifications@github.com> wrote:

> In lib/puppet/provider/user/hpux.rb
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22981828>:
>
> >
> >    def deletecmd
> >      super.insert(1,"-F")
> >    end
> >
> >    def modifycmd(param,value)
> > -    super.insert(1,"-F")
> > +     cmd = super(param, value)
> > +     cmd << "-F"
> > +     if self.trusted == "Trusted"
> > +        cmd << ";"
>
> Thanks. The self.trusted is the part I have not been able to mock out in a
> spec file.
> … <#14aeaea986607561_>
> On Jan 14, 2015 7:04 PM, "Michael Smith" <notifications@github.com>
> wrote: In lib/puppet/provider/user/hpux.rb <
> https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22981701>:
> > > def deletecmd > super.insert(1,"-F") > end > > def
> modifycmd(param,value) > - super.insert(1,"-F") > + cmd = super(param,
> value) > + cmd << "-F" > + if self.trusted == "Trusted" > + cmd << ";" I'll
> see about coming up with specs that work for this. — Reply to this email
> directly or view it on GitHub <
> https://github.com/puppetlabs/puppet/pull/3443/files#r22981701>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22981828>.
>